security_privacy_ln.asciidoc

```diff
@@ -0,0 +1,614 @@
+More precisely, a payment channel cannot pertain to a payment if the remaining time the payment might be locked for is larger than what the forwarding node would be willing to accept.
+This leakage could be evicted by adhering to the so-called shadow routes.
+
+One of the most subtle and yet powerful leakages an on-path adversary can foster is the timing analysis.
+An on-path adversary can log for every routed payment along with the amount of time it takes for a node to respond to an HTLC request.
+Before starting the attack, the attacker learns every node's latency characteristics in the Lightning Network by sending them requests.
+Naturally, this can aid in establishing the adversary's precise position in the payment path.
+Even more, as it was recently shown, an attacker can successfully determine the sender and the receiver of a payment from a set of possible senders and receivers using time-based estimators.
+
+Finally, we remark that several, yet unknown, leakages might exist that can aid deanonymizing attempts.
+For instance, even knowing the applied routing algorithm could help exclude certain nodes from being a sender and/or receiver of a payment.
+We note that different Lightning wallets apply different routing algorithms.
+Likely, many more leakages exist.
+
+== Revealing channel balances (probing)
+
+The balances of Lightning channels are supposed to be hidden for privacy and efficiency reasons.
+A Lightning node only knows the balances of its adjacent channels.
+The protocol provides no standard way to query the balance of a remote channel.
+
+However, an attacker can reveal the balance of a remote channel in a _probing attack_.
+In information security, probing refers to the technique of sending requests to a targeted system and making conclusions about its private state based on the received responses.
+
+Lightning channels are prone to probing.
+Recall that a standard Lightning payment starts with the receiver creating a random payment secret and sending its hash to the sender.
+Note for the intermediary nodes, all hashes look random.
+There is no way to tell whether a hash corresponds to a real secret or was generated randomly.
+
+The probing attack proceeds as follows.
+Say, the attacker Mallory wants to reveal Alice's balance of a public channel between Alice and Bob.
+Suppose the total capacity of that channel is 1 million satoshis.
+Alice's balance could be anything from zero to 1 million satoshis (to be precise, the estimate is a bit tighter due to channel reserve, but we don't account for it here for simplicity).
+Mallory opens a channel with Alice with 1 million satoshis and sends 500 thousand satoshis to Bob via Alice using a _random number_ as the payment hash.
+Of course, this number does not correspond to any known payment secret.
+Therefore, the payment will fail.
+The question is: how exactly will it fail?
+
+There are two scenarios.
```

Comment:
Simplified example, but this is missing a bit of nuance re the existence of the channel level flow control constraints that may be slightly out of sync with the advertised routing constraints. As an example, let's say a channel only allows 5 HTLCs for w/e reason in this case when a prober attempts to attach a 6th HTLC, they'll get a "temp chan failure" even though the channel may have had sufficient bandwidth to carry the channel. 

Another value to consider is the `max_htlc` setting, which may not allow a prober to deduce the full available bandwidth amount in the channel in a single go.